optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/config/SolverBenchmarkConfig.java

```diff
@@ -101,6 +117,9 @@ public void inherit(SolverBenchmarkConfig inheritedConfig) {
         } else if (inheritedConfig.getProblemBenchmarksConfig() != null) {
             problemBenchmarksConfig.inherit(inheritedConfig.getProblemBenchmarksConfig());
         }
+        if (subSingleCount == null) {
+            subSingleCount = ObjectUtils.defaultIfNull(inheritedConfig.getSubSingleCount(), 1);
```

Comment:
Reading an Config object and then writing it again should not alter the XML. This code will do that.

This is better:
if (subSingleCount == null) {
            subSingleCount = inheritedConfig.getSubSingleCount();
}
But there's a ConfigUtils.method to do that for you.

Don't do defaulting in the inherit() method, do it in locally in the build methods (so it doesn't change the global fields of a config class.